js/jobmixins.js

```diff
@@ -354,13 +358,14 @@ exports.do_download = function(op, callback) {
  * encounter a collision.
  */
 var saveToDeviceStorage = exports.saveToDeviceStorage =
-function(_LOG, blob, storeTo, filename, partInfo, cb, isRetry) {
+function(_LOG, blob, storeTo, registerDownload, filename, partInfo, cb,
+         isRetry) {
```

Comment:
Whoops, good eye!  And this did actually break three separate POP3 tests; I was remiss in doing a full run before putting this up for review.  Apologies and I'll fix this now.